packages/wonder-blocks-dropdown/components/dropdown-launcher.js

```diff
@@ -0,0 +1,244 @@
+            return;
+        }
+
+        if (selectedValues.includes(selectedValue)) {
+            const index = selectedValues.indexOf(selectedValue);
+            const updatedSelection = [
+                ...selectedValues.slice(0, index),
+                ...selectedValues.slice(index + 1),
+            ];
+            onChange(updatedSelection);
+        } else {
+            // Item was newly selected
+            onChange([...selectedValues, selectedValue]);
+        }
+        this.handleItemSelected();
+    };
+
+    _openDropdown = () => {
+        this.setState({open: true});
+    };
+
+    _getMenuItems = (): Array<DropdownItem> => {
+        const {selectedValues} = this.props;
+        return React.Children.toArray(this.props.menuItems)
+            .filter(Boolean)
+            .map((item) => {
+                const {value, disabled} = item.props;
+                const itemObject = {
+                    component: item,
+                    focusable:
+                        ActionItem.isClassOf(item) || OptionItem.isClassOf(item)
+                            ? !disabled
+                            : false,
+                    populatedProps: {},
+                };
+                if (ActionItem.isClassOf(item)) {
+                    return {
+                        ...itemObject,
+                        populatedProps: {
+                            indent: true,
+                            onClick: this.handleItemSelected,
+                        },
+                    };
+                } else if (OptionItem.isClassOf(item)) {
+                    return {
+                        ...itemObject,
+                        populatedProps: {
+                            onToggle: this.handleOptionSelected,
+                            selected: selectedValues
+                                ? selectedValues.includes(value)
+                                : false,
+                            variant: "check",
+                        },
+                    };
+                } else {
+                    return itemObject;
+                }
+            });
+    };
+
+    handleOpenerRef = (node: any) => {
+        this.openerElement = ((ReactDOM.findDOMNode(node): any): HTMLElement);
+    };
+
+    render() {
+        const dropdownItems = this._getMenuItems();
+        const {
+            children,
+            alignment,
+            disabled,
+            style,
+            dropdownStyle,
+            /* eslint-disable no-unused-vars */
+            onChange,
+            selectedValues,
+            "aria-disabled": ariaDisabled,
+            ...sharedProps
+        } = this.props;
+
+        const opener = (
+            <span
+                ref={this.handleOpenerRef}
+                disabled={!dropdownItems.length || disabled}
+            >
+                {children(this._openDropdown)}
+            </span>
+        );
+
+        return (
+            <Dropdown
```

Comment:
`Dropdown` is really starting to feel like it should be called `DropdownCore` and `DropdownLauncher` should probably be renamed to `DropdownCore` or simple `Dropdown`.  Don't worry about doing this in this PR but something to consider when writing up the implementation spec.

Reply:
@kevin Agree on this, we need to provide an API that feels more natural for devs that use these components. I'm also thinking a lot about this for the `Popover`. As these components (dropdown and popover) have some similarities, maybe this is a good point to agree on this API for these components for consistency purposes (including Modal as well). What do you think about this?

### Dropdown
- `Dropdown` becomes `DropdownCore`.
- `DropdownLauncher` becomes `Dropdown`.

### Modal
- `ModalLauncher` becomes `Modal`.

### Popover
- `Popover` could be the launcher as well.
- `PopoverCore`: Base popover that is used by all the variations. Similar to what `Dropdown` is doing as of today.

Reply:
The modal launcher change kind of makes sense since we're calling things dialogs now in v2.  Thanks for think about how pattern might work across similar components.